tests/e2e/test_decorators.py

```diff
@@ -0,0 +1,37 @@
+"""Test decorators."""
+
+import time
+from functools import wraps
+
+
+def retry(max_attempts=3, wait_between_runs=5):
+    """Construct decorator that allows to retry running selected test."""
+
+    def retry_test_decorator(test_function):
+        @wraps(test_function)
+        def wrapper(*args, **kwargs):
+            retry_count = 1
+
+            while retry_count < max_attempts:
+                try:
+                    # try to run the test
+                    return test_function(*args, **kwargs)
```

Comment:
@tisnik  can we condition these retries so we're only retrying it on particular expected failures?

Reply:
we can, I'll do update